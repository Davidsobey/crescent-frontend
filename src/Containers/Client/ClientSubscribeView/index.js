/**
 *
 * SubscriptionView
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { reduxForm } from 'redux-form';
import ReactTable from 'react-table';
import 'react-table/react-table.css';
import Tooltip from 'material-ui/Tooltip';
import { CircularProgress } from 'material-ui/Progress';

import history from '../../../Helpers/History';
import Card from '../../../Components/Card';
import ClientActions from '../../../Actions/ClientActions';
import CourseActions from '../../../Actions/CourseActions';
import IconButton from '../../../Styles/IconButton';
import CustomModal from '../../../Components/Modal/index';
import { StyledDelete } from '../../../Styles/Delete';
import { StyledEdit } from '../../../Styles/Edit';

class SubscriptionView extends React.Component {
  constructor(props) {
    super(props);
    this.state = { obj: {} };
    this.props.dispatch(CourseActions.loadCoursesByClientSubscriptions(props.user.clientId));
    this.props.dispatch(ClientActions.getUserEnrolments(this.props.user.clientId));

    this.manipulateData = this.manipulateData.bind(this);
  }

  
  manipulateData = (courses, userEnrolments) => {
    var getEnrolmentsCount = courseId =>
    Array.isArray(userEnrolments) ? userEnrolments.filter(userEnrolment => userEnrolment.courseId == courseId).length : '';

    const data = [];
    if (Array.isArray(courses)) {
      courses
      .forEach((course) => {
        const row = {
          courseName: course.name,
          courseDescription: course.description,
          enrolmentsCount: getEnrolmentsCount(course.id),
          price: course.price,
        };
        data.push(row);
      });
    }
    return data;
  };

  render() {
    const columns = [
      {
        Header: 'Course Name',
        accessor: 'courseName',
      },
      {
        Header: 'Course Description',
        accessor: 'courseDescription',
      },
      {
        Header: 'Enrolments Count',
        accessor: 'enrolmentsCount',
      },
      {
        Header: 'Price',
        accessor: 'price',
      },
    ];
    return (
      <div>
        <Card width="800px" title="Client List">
          {this.props.subscribedCourses_loading || this.props.userEnrolments_loading ? (
            <div className="center">
              <CircularProgress color="secondary" />
            </div>
          ) : (
            <div>
              <ReactTable
                columns={columns}
                data={this.manipulateData(this.props.subscribedCourses, this.props.userEnrolments)}
                filterable
                defaultPageSize={10}
                className="-striped -highlight"
              />
            </div>
          )}
        </Card>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  subscribedCourses: state.CourseReducer.subscribedCourses,
  subscribedCourses_loading: state.CourseReducer.subscribedCourses_loading,
  userEnrolments: state.ClientReducer.userEnrolments,
  userEnrolments_loading: state.ClientReducer.loading,
  user: state.LoginReducer.user,
});

const withForm = reduxForm(
  {
    form: 'subscriptionView',
  },
  SubscriptionView,
);

SubscriptionView.propTypes = {
  dispatch: PropTypes.func,
  subscribedCourses: PropTypes.array,
  subscribedCourses_loading: PropTypes.bool,
  userEnrolments: PropTypes.array,
  userEnrolments_loading: PropTypes.bool,
  user: PropTypes.object,
};

export default compose(connect(mapStateToProps), withForm)(SubscriptionView);
