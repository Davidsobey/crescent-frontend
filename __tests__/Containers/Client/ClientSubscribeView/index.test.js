import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import ClientSubscribeView from '../../../../src/Containers/Client/ClientSubscribeView/index.js';


const mockStore = configureStore();
const initialState = {
  ClientReducer:{
    userEnrolments: [],
    loading: false,
  },
  CourseReducer:{
    subscribedCourses: [],
    subscribedCourses_loading: false,
  },
  LoginReducer: {
    user: {
      name: 'aaa',
    }
  },
};
const store = mockStore(initialState);

describe('<ClientSubscribeView />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<ClientSubscribeView store={store} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});