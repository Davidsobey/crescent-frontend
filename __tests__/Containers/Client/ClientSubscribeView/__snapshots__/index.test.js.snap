// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientSubscribeView /> render() renders the component 1`] = `
<Connect(Form(SubscriptionView))
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="subscriptionView"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  subscribedCourses={Array []}
  subscribedCourses_loading={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  user={
    Object {
      "name": "aaa",
    }
  }
  userEnrolments={Array []}
  userEnrolments_loading={false}
/>
`;
